src/diff/children.js

```diff
@@ -205,41 +196,41 @@ export function diffChildren(
 			//
 			// To fix it we make sure to reset the inferred value, so that our own
 			// value check in `diff()` won't be skipped.
-			if (internal.type === 'option') {
+			if (parentInternal.type === 'option') {
 				// @ts-ignore We have validated that the type of parentDOM is 'option'
 				// in the above check
 				parentDom.value = '';
 			}
 		} else if (
 			startDom &&
 			childInternal != null &&
-			childInternal._dom == startDom &&
+			prevDom == startDom &&
 			startDom.parentNode != parentDom
 		) {
 			// The above condition is to handle null placeholders. See test in placeholder.test.js:
 			// `efficiently replace null placeholders in parent rerenders`
-			startDom = getDomSibling(childInternal);
+			startDom = nextDomSibling; // Note: change works in restructure. Not dependent on backing tree
```

Comment:
This change isn't related to backing nodes and works on `restructure` directly. Just including it here since it was an easy change to make